### PR TITLE
Added Dockerfile for virtual Robocup 2021

### LINKED ISSUE #2011: Docker Image for Robocup 2021
### Description of the task
According to https://github.com/RoboCup-SSL/ssl-simulation-setup, we need to use docker to manage our dependencies and forward gui over vnc to compete in the virtual robocup. 

### Acceptance criteria

- [ ] docker image for robocup 2021

### PR DESCRIPTION
**Pull request by PhilD71**
### Description
Added Dockerfile for virtual Robocup 2021. 

### Testing Done
Tested with the SSL-Simulation-Setup, the same software we will be using at robocup

### Resolved Issues
resolves #2011 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

A PR was opened in the SSL-Simulation-Setup repository as well to add our dockerfile to the setup. [link to PR](https://github.com/RoboCup-SSL/ssl-simulation-setup/pull/2)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Docker file for virtual robocup 2021
+
+FROM robocupssl/ubuntu-vnc:latest
+
+# Creates a directory called scripts, copies
+# getRepo script and runs it to fetch the git
+# repository. Once the repository is fetched,
+# it runs setup_software
```
Comment (akhilveeraghanta) [inline docker/Dockerfile]: this comment just needs to be updated

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm! just one comment from my end

thoughts on putting the `Dockerfile` in the `environment_setup` folder?

File: docker/Dockerfile (hunk 1 above)
Comment (PhilD71) [reply docker/Dockerfile]: I forgot to save it in vscode before committing! Should be updated now!

**Comment by PhilD71** [comment]
@akhilveeraghanta I was planning on adding a second image with the development docker setup, but I'll get to that after robocup as its not as important. I do agree, probably best to place it in env setup to keep root tidy

**Review by jonathanlew — COMMENTED** [review]
Can we add some documentation in the docs folder?

File: docs/getting-started.md [hunk 2]
```diff
@@ -245,3 +251,21 @@ This will output the file at the _absolute_ path given via the `--callgrind-out-
...
+
+### Setting up the SSL Simulation Environment
+
+1. Fork the [SSL-Simulation-Setup](https://github.com/RoboCup-SSL/ssl-simulation-setup) repository.  
+2. Clone it.
+3. Follow these [instructions](https://github.com/RoboCup-SSL/ssl-simulation-setup/blob/master/Readme.md) to set up and run the repository.
+
+### Pushing a Dockerfile to dockerhub
+
+After editing the dockerfile, you build the image and push it to dockerhub with the following steps
```
Comment (jonathanlew) [inline docs/getting-started.md]: nit:
```suggestion
After editing the dockerfile, build the image and push it to dockerhub with the following steps
```

File: docs/getting-started.md [hunk 3]
```diff
@@ -245,3 +251,21 @@ This will output the file at the _absolute_ path given via the `--callgrind-out-
 2. To regenerate code from the `.ioc` file, run `bazel run //firmware_new/tools:cubemx_regen path/to/directory/with/.ioc`. The directory that is passed in as an argument must only contain 1 ioc file, which will be used to generate code into the same directory.
 
 To make sure we are all using the same cube version, run `STM32CubeMX` when editing the `.ioc` file.
+
+## Setting up virtual robocup 2021
```
Comment (jonathanlew) [inline docs/getting-started.md]: nit: 
```suggestion
## Setting up Virtual Robocup 2021
```

**Review by jonathanlew — COMMENTED** [review]
couple nits

**Review by jonathanlew — APPROVED** [review]
:ship: :anchor:

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**